src/date-picker/calendar/index.tsx

```diff
@@ -109,18 +120,20 @@ const Calendar = React.forwardRef(
     }, [baseDate, focusedDate, gridHasFocus]);
 
     useEffect(() => {
-      const calendarShouldHaveFocus = calendarHasFocus;
       const calendarActuallyHasFocus = elementRef.current?.contains(document.activeElement);
 
-      if (calendarShouldHaveFocus && !calendarActuallyHasFocus) {
+      if (!calendarActuallyHasFocus) {
         elementRef.current?.focus();
```

Comment:
The calendar is focused when the focus goes away from the grid. This is actually not needed. I removed this useEffect, onGridBlur and onGridFocus and everything works as expected in the default date picker. All tests are passing. Can you confirm?

I think this code was to provide a tab trapping logic, but then another tab trap solution was added w/o proper cleanup.

Reply:
Can confirm behaviour is the same. Removing these also fixes the other issue with the focusable element not focusable after the date picker.